src/services/weather.js

```diff
@@ -30,6 +34,28 @@ async function fetchWeatherData(city) {
   return mapModelToView(response.data);
 }
 
-function getIconURL(icon) {
+export function getIconURL(icon) {
   return weatherImageURL + icon + imageExtension;
 }
+
+function mapModelForecastToView(model) {
+  return _.map(model, mapModelToView);
+}
+
+export const fetchWeatherForcastForCity = async (city, completation) => {
+  const response = await axios.get('https://api.openweathermap.org/data/2.5/forecast', {
+    params: {
+      q: city,
+      APPID: Config.APPID,
+    },
+  });
+
+  completation(
```

Comment:
about this `completation`...
Why not simply return `forecastData` and then await the result and use it?
I'm aware that we already talked about it, I think you were trying to avoid async `componentDidMount`? We can talk about it and find a better solution. Ping me when you get to this :)
 
But it is definitely better to just return data out of this function, it's what somebody who'll use it would expect.

Reply:
Also, did you ever use `InteractionManager` before? https://facebook.github.io/react-native/docs/interactionmanager.html

It's something that'll definitely improve performances for this case.
What we're trying to avoid is that component is waiting for server data and doesn't get properly rendered because of it. 
In such cases, you can wrap your fetch in `InteractionManager.runAfterInteractions`. That way, first all the UI stuff is done, you don't have any twitches in animations (when switching screens, for instance) and only after that, js thread will go on to fetch the data. 

For your app it probably won't make any visible difference but should it grow to something more complex, this is definitely a good practice.